src/ik_parallel.h

```diff
@@ -195,27 +195,27 @@ struct IKParallel
             double current_goal_fitness = solvers[i]->computeFitness(result, fk.getTipFrames());
             if (current_goal_fitness)
             {
-                ROS_DEBUG_STREAM("Calculated fitness for thread: " << i);
+                ROS_DEBUG_STREAM("Calculated fitness for thread: " << i << " is: " << current_goal_fitness);
             }
-            if (current_goal_fitness < previous_solver_fitness_[i])
+            if (success || current_goal_fitness < previous_solver_fitness_[i])
```

Comment:
What's happening here exactly? I would expect `success && current_goal_fitness < previous_solver_fitness_[i]` instead

Reply:
Is it the case that this is updating with the best solution so far untill a completely successful one is available, at which point ths thread exits and the final successful solution is used? In which case this syntax is correct, if a little opaque.